src/guacd/daemon.c

```diff
@@ -304,8 +304,23 @@ int main(int argc, char* argv[]) {
 
     }
 
+    /* create socket for AF_INET6 to allow IPv6 bind in case there is no IPv4 address available */
+    sa_family_t family = AF_INET6;
+    current_address = addresses;
+    while (current_address != NULL) {
+
+        if (current_address->ai_family == AF_INET)
+        {
+            family = AF_INET;
+        }
+
+        current_address = current_address->ai_next;
+    }
```

Comment:
I don't think it is strictly correct to have two loops through the possible addresses like this - one to choose the family and then another to choose the address (presumably failing repeatedly for if that family was pulled from a different address). Shouldn't this all be done in the same loop?

Reply:
I had a similar thought, but initially didn't feel like changing the proven flow of control much.

It would probably be OK to re-create the socket a few times (we set `SO_REUSEADDR`). I re-factored the code accordingly.

Please review my updated (force-pushed) commit.

Thank you!